nemo/collections/tts/parts/utils/tts_dataset_utils.py

```diff
@@ -150,3 +175,55 @@ def common_path(path1, path2):
         base_dir = common_path(base_dir, audio_dir)
 
     return base_dir
+
+
+def filter_dataset_by_duration(entries: List[Dict[str, Any]], min_duration: float, max_duration: float):
+    """
+    Filter out manifest entries based on duration.
+
+    Args:
+        entries: List of manifest entry dictionaries.
+        min_duration: Minimum duration below which entries are removed.
+        max_duration: Maximum duration above which entries are removed.
+
+    Returns:
+        filtered_entries: List of manifest entries after filtering.
+        total_hours: Total duration of original dataset, in hours
+        filtered_hours: Total duration of dataset after filtering, in hours
+    """
+    filtered_entries = []
+    total_duration = 0.0
+    filtered_duration = 0.0
+    for entry in entries:
+        duration = entry["duration"]
+        total_duration += duration
+        if (min_duration and duration < min_duration) or (max_duration and duration > max_duration):
+            continue
+
+        filtered_duration += duration
+        filtered_entries.append(entry)
+
+    total_hours = total_duration / 3600.0
+    filtered_hours = filtered_duration / 3600.0
+
+    return filtered_entries, total_hours, filtered_hours
+
+
+def get_weighted_sampler(
+    sample_weights: List[float], batch_size: int, num_steps: int
+) -> torch.utils.data.WeightedRandomSampler:
+    """
+    Create pytorch sampler for doing weighted random sampling.
+
+    Args:
+        sample_weights: List of sampling weights for all elements in the dataset.
```

Comment:
Ordering is I guess the same as the manifest? What happens if some entries are filtered out?

Reply:
Do we expect the user to set the sample weights? If so, we would want the correspondence between index and specific entry to be as clear as possible.

Mostly I'm worried about a situation where sample weights are set for the dataset in manifest order (since `self.data_samples` is constructed by reading through the manifest), but then some entry is skipped that the user didn't expect.

Reply:
In this data loader implementation the user only specifies how much weight to give to each training manifest (eg. in the usage example, 20% to LJSpeech and 80% to VCTK). Creating the item level sample weights (where are all items in the same manifest receive the same weight) is done internally.

We could also allow user to provide a "sample_weight" field in each row in the training manifest...but I could not think of any practical use case where someone would use that over the manifest-level weighting.

Unless your concern is that someone might use this utility function directly when defining their own data loader. In which case, it is a little error prone.

Reply:
Ahhh. Alright, that makes sense to me 👍